docs/kb/pcr-measurements.md

```diff
@@ -0,0 +1,157 @@
+# PCR measurements performed by Dasharo firmware
+
+## Introduction
+
+As part of measured boot process firmware hashes (measures) various
+pieces of code or data and updates PCRs of a TPM device.  This allows a user
+(typically, automatically) to attest integrity of the system by verifying PCR
+values after a boot or tying decryption of data to expected values.
+
+Usually the firmware is not the only entity which updates PCRs and even if it
+is, the number of updates can be large, so it's important to know which values
+do get measured to which PCRs and under what conditions (e.g., some
+measurements are done in response to user actions).
+
+TPM event log maintained throughout the boot process is meant to track this kind
+of information, but due to limitations of its format the log is rarely enough
+to understand all of the measurements.  This document is meant to describe when
+Dasharo updates PCRs so users would know which PCRs to use and when to expect
+their values to be changed.
+
+Firmware components are described separately because not all components might
+be present in a given firmware variant.
+
+## The truth about PCR usage
+
+[TPM 1.2 Specification][TPM12] (section 3.3.3) and
+[TPM 2.0 Specification][TPM20] (section 3.3.4) provide information on which
+PCRs are expected to be used for what purposes.  The problem is that
+instructions seemingly contradict themselves and even if when they don't,
+discerning which PCR is to be used for a particular code or data is far from
+an easy task.  This results in a disconnect between what's prescribed by those
+specification and how things are in practice.
+
+EDK is better at adhering to TCG specifications probably because of interactions
+between developers on both sides.  The way coreboot updates PCRs originally came
+from Chromebooks and its authors had a different interpretation of PCR
+allocations ([see][coreboot-pcrs]).  When EDK is used as a second-stage
+firmware after coreboot, you get a mix of both.
+
+[TPM12]: https://trustedcomputinggroup.org/wp-content/uploads/TCG_PCClientImplementation_1-21_1_00.pdf
+[TPM20]: https://trustedcomputinggroup.org/wp-content/uploads/TCG_PCClient_PFP_r1p05_v23_pub.pdf
+[coreboot-pcrs]: https://doc.coreboot.org/security/vboot/measured_boot.html#platform-configuration-registers
+
+## coreboot measurements
+
+### PCR banks
+
+At the time of writing (June 2024) coreboot has no support for multiple PCR
+banks but allows choosing a hash algorithm for TPM 2.0 log format.  This choice
```

Comment:
I don't like the wording, multiple PCR banks (which don't really have anything to do with event log) can be enabled by coreboot, it just doesn't make use of them. One could argue that this poses a security issue, which EDK (kind of) deals with, see next comment.

Reply:
https://github.com/Dasharo/docs/compare/d8644747bf6d813d99ac1d9f52c60a916ef47475..c0376d7b5c8c64f9124139a5bc44c20cdf472812#diff-c2a55b23fc3bd588ed98d40c36104af5b9175c3acdc2ab8c0288d1590992e689R48